flink-table/flink-table-common/src/test/java/org/apache/flink/table/types/extraction/TypeInferenceExtractorTest.java

```diff
@@ -1810,4 +1844,30 @@ public String eval(String f1, String f2) {
             return "";
         }
     }
+
+    private static class ArgumentHintScalarFunctionNotNullTypeWithOptionals extends ScalarFunction {
+        @FunctionHint(
+                argument = {
+                    @ArgumentHint(
+                            type = @DataTypeHint("STRING NOT NULL"),
+                            name = "f1",
+                            isOptional = true),
+                    @ArgumentHint(type = @DataTypeHint("INTEGER"), name = "f2", isOptional = true)
+                })
+        public String eval(String f1, Integer f2) {
```

Comment:
After verification, there will be a NullPointerException during execution. I have added relevant detection and judgment, but this may not only be an optional issue. If the primitive is declared as a nullable type, there may be similar problems, but currently there is no similar detection. However, in order to avoid modifying the existing Function temporarily, I have not made any changes to this part. In the future, a separate JIRA can be created to track this issue.